x/mongo/driver/topology/pool.go

```diff
@@ -83,33 +83,22 @@ func connectionExpiredFunc(v interface{}) bool {
 	if !ok {
 		return true
 	}
-	var reason string
-	disconnected := atomic.LoadInt32(&c.pool.connected) != connected
-	if disconnected {
-		reason = event.ReasonPoolClosed
-	}
-
-	idle := c.expired()
-	if !disconnected && idle {
-		reason = event.ReasonIdle
-	}
-
-	stale := c.pool.stale(c)
-	if !disconnected && !idle && stale {
-		reason = event.ReasonStale
-	}
 
-	res := disconnected || stale || idle
-	if res && c.pool.monitor != nil {
-		c.pool.monitor.Event(&event.PoolEvent{
-			Type:         event.ConnectionClosed,
-			Address:      c.pool.address.String(),
-			ConnectionID: c.poolID,
-			Reason:       reason,
-		})
-	}
-
-	return res
+	switch {
+	case atomic.LoadInt32(&c.pool.connected) != connected:
```

Comment:
Can we add a method to the pool like we did for the connection to indicate whether it's closed?

`case c.pool.closed():`